src/views/Proposal.vue

```diff
@@ -25,7 +28,32 @@ const discussion = computed(() => {
   return output;
 });
 
+const metadataUrl = computed(() => {
+  if (!proposal.value) return null;
+
+  return getUrl(proposal.value.metadata_uri);
+});
+
+async function fetchMetadata() {
+  const res = await fetch(metadataUrl.value);
```

Comment:
Is this just to get the execution data? It would be better to do index it on sx-api to avoid extra request to IPFS in the frontend. These request can be slow

Reply:
Thanks. I didn't even notice it was already in sx-api.